test/common/index.js

```diff
@@ -102,6 +102,9 @@ if (process.argv.length === 2 &&
 
 const isWindows = process.platform === 'win32';
 const isAIX = process.platform === 'aix';
+// On IBM i, process.platform and os.platform() both return 'aix',
+// It is not enough to differentiate between IBM i and real AIX system.
+const isIBMi = isAIX && execSync('uname -s').toString() === 'OS400\n';
```

Comment:
Even with the `isAIX` guard, I think we should try to avoid doing this. Can you confirm what `os.type()` yields on IBM i.

Reply:
I agree that avoiding doing this every time would be good. If we don't already have enough info to figure it out maybe adding something to process for IBMi would make sense?  I'm guessing we may want to know in other places as well.

Reply:
`os.type()` is built on `uv_os_uname()`. If `uv_os_uname()` doesn't already identify IBM i somehow, we could update it.

Reply:
@dmabupt you can switch on `os.type()` as @cjihrig suggested:
```
core/node (master $ u=) % ssh iojs@ibmi-1
-bash-4.4$ cd node
-bash-4.4$ ./out/Release/node -p 'require("os").type()'
OS400
-bash-4.4$ uname -a
OS400 iOSSBld1 3 7 00210009F22V Os
```
aix:
```
-bash-4.3$ ./out/Release/node -p 'require("os").type()'
AIX
-bash-4.3$ uname -a
AIX power8-nodejs3 1 6 00FA74164C00
```

Reply:
done.